tools/run_tests.py

```diff
@@ -67,50 +67,72 @@ def __download_cli_to_sandbox(self):
         return cli_filepath
 
 
-    def __configure_cli(self, sandbox_env, dcos_url):
+    def __configure_cli(self, dcos_url):
         cmds = [
+            'which dcos',
             'dcos config set core.dcos_url "{}"'.format(dcos_url),
             'dcos config set core.reporting True',
             'dcos config set core.ssl_verify false',
             'dcos config set core.timeout 5',
             'dcos config show']
         for cmd in cmds:
-            subprocess.check_call(cmd.split(' '), env=self.__sandbox_env)
+            subprocess.check_call(cmd.split(' '))
 
 
     def setup_cli(self):
         try:
             self.__github_updater.update('pending', 'Setting up CLI')
-            self.__sandbox_env = self.__get_cli_sandbox()
+            self.__configure_cli_sandbox() # adds to os.environ
             cli_filepath = self.__download_cli_to_sandbox()
-            self.__configure_cli(self.__sandbox_env, self.__dcos_url)
+            self.__configure_cli(self.__dcos_url)
             dcos_login.DCOSLogin(self.__dcos_url).login()
         except:
             self.__github_updater.update('error', 'CLI Setup failed')
             raise
 
 
     def run_shakedown(self, test_dirs, requirements_file = ''):
+        # keep virtualenv in a consistent/reusable location:
+        if 'WORKSPACE' in os.environ:
+            virtualenv_path = os.path.join(os.environ['WORKSPACE'], 'env')
+        else:
+            virtualenv_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'env')
+        logger.info('Configuring virtualenv in {}'.format(virtualenv_path))
+        # to ensure the 'source' call works, just create a shell script and execute it directly:
+        script_path = os.path.join(self.__sandbox_path, 'run_shakedown.sh')
+        script_file = open(script_path, 'w')
         if requirements_file:
-            cmd = 'pip install --no-cache-dir -r {}'.format(requirements_file)
-            try:
-                self.__github_updater.update('pending', 'Configuring test prerequisites')
-                subprocess.check_call(cmd.split(' '), env=self.__sandbox_env)
-            except:
-                self.__github_updater.update('error', 'Test prerequisites failed')
-                raise
-        cmd = 'shakedown --dcos-url "{}" --ssh-key-file none {}'.format(self.__dcos_url, test_dirs)
+            requirements_cmd = 'pip install -r {}'.format(requirements_file)
+        else:
+            requirements_cmd = ''
+        script_file.write('''
+#!/bin/bash
+set -e
+echo "VIRTUALENV CREATE/UPDATE: {venv_path}"
+virtualenv -p $(which python3) --always-copy {venv_path}
+echo "VIRTUALENV ACTIVATE: {venv_path}"
+source {venv_path}/bin/activate
+echo "REQUIREMENTS INSTALL: {reqs_file}"
+{reqs_cmd}
+echo "SHAKEDOWN RUN: {test_dirs}"
+shakedown --dcos-url "{dcos_url}" --ssh-key-file "" --stdout=all --stdout-inline {test_dirs}
```

Comment:
Added TODOs to remove the inlined scripts (either with external templating, or with a better method to invoke the virtualenv)